### PR TITLE
Get route prices from quoter

### PR DESCRIPTION
**Pull request by PierreMkt**
instead of hardcoding pair prices

### CODE REVIEW

**Review by michcio1234 — APPROVED** [review]
Nice! Not sure I like the idea to put it in `quoters`, but I don't have a strong opinion, so we can keep it.

File: src/models.ts [hunk 1]
```diff
@@ -38,6 +38,8 @@ export class RouteSpec {
     public maxSellAmount: bigint;
     /** The L1 block number for which the route must be valid */
     public blockNumber: bigint;
+    /** The price of the pair */
+    public pairPrice?: number;
```
Comment (michcio1234) [inline src/models.ts:42]: That's a weird idea to put it here :P

**Review by michcio1234 — CHANGES_REQUESTED** [review]
I will change it.

File: src/solvers/uniswap/uniswapSolver.ts [hunk 2]
```diff
@@ -120,4 +120,62 @@ export class UniswapSolverAdapter extends SolverAdapter {
...
+            const price =
+                buyToken.fromOnchainAmount(BigInt(buyAmount)) /
+                sellToken.fromOnchainAmount(routeSpecs[index].minSellAmount);
+            this.setRouteSpecPairPrice(routeSpecs[index], price);
+        });
+    }
+
+    /**
+     * Calls the Propeller quoter API to retrieve the quote for given route specs.
+     * @returns A promise that resolves to a list of quote mappings .
```
Comment (michcio1234) [inline src/solvers/uniswap/uniswapSolver.ts:153]: What does it mean? Mappings from what to what?

File: src/solvers/uniswap/uniswapSolver.ts [hunk 3]
```diff
@@ -9,29 +9,36 @@ import { SolverAdapter } from "../solverAdapter";
...
-    supportedTokens,
-    tokenPairPrices,
-    usv3PoolFee,
-} from "./constants";
+import { UNIV3_ROUTER, usv3PoolFee } from "./constants";
+import axios from "axios";
+import { ADDR2TOKEN, NULL_ADDRESS } from "../../constants";
+import { Quote } from "./models";
+
+const QUOTER_API_URL = "https://dev-api.propellerheads.xyz/v1/solver/turbine/quote/";
```
Comment (carloszanella) [inline src/solvers/uniswap/uniswapSolver.ts:17]: are we keeping this URL? does it work (is it public)?

File: __tests__/testIntegration.test.ts [hunk 4]
```diff
@@ -83,17 +82,25 @@ describe("SolverAdapter", () => {
...
+                expect(minQuote.sell_token).toBe(spec.sellToken);
+                expect(maxQuote.sell_token).toBe(spec.sellToken);
+                expect(minQuote.buy_token).toBe(spec.buyToken);
+                expect(maxQuote.buy_token).toBe(spec.buyToken);
+                expect(BigInt(minQuote.sell_amount)).toBeGreaterThanOrEqual(
+                    spec.minSellAmount
                 );
-                expect(BigInt(quote.sell_amount)).toBeLessThanOrEqual(
-                    routeSpecs[index].maxSellAmount
+                expect(BigInt(maxQuote.sell_amount)).toBeLessThanOrEqual(
```
Comment (michcio1234) [inline __tests__/testIntegration.test.ts:98]: Why do we allow it to be **less**?

File: __tests__/testIntegration.test.ts [hunk 5]
```diff
@@ -83,17 +82,25 @@ describe("SolverAdapter", () => {
...
+                    spec.maxSellAmount
                 );
-                expect(quote.buy_amount).toBeDefined();
+                expect(minQuote.buy_amount).toBeDefined();
+                expect(maxQuote.buy_amount).toBeDefined();
```
Comment (michcio1234) [inline __tests__/testIntegration.test.ts:102]:
```suggestion
                expect(maxQuote.buy_amount).toBeDefined();
                expect(maxQuote.buy_amount).toBeGreaterThan(minQuote.buy_amount);
```

File: src/solvers/uniswap/models.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        buy_token: Address,
+        sell_amount: bigint,
+        buy_amount: bigint,
+    ) {
+        this.sell_token = sell_token;
+        this.buy_token = buy_token;
+        this.sell_amount = sell_amount;
+        this.buy_amount = buy_amount;
+    }
+}
```
Comment (michcio1234) [inline src/solvers/uniswap/models.ts:26]: I don't mind having this, but sometimes interfaces are nicer (more concise):
```suggestion
export interface Quote {
    /** The token to sell */
    public sell_token: Address;
    /** The token to buy */
    public buy_token: Address;
    /** The minimum amount of the sell token the route must handle */
    public sell_amount: bigint;
    /** The maximum amount of the sell token the route must handle */
    public buy_amount: bigint;
}
```

**Review by michcio1234 — APPROVED** [review]

**Pull request merged**